app/src/main/java/net/squanchy/settings/SettingsFragment.java

```diff
@@ -75,6 +90,8 @@ private void removeDebugCategory() {
     public void onStart() {
         super.onStart();
 
+        locationPreferences.setChecked(optInPreferencePersister.getOptInPreferenceGranted());
```

Comment:
Here would also be the place where you check the remote config and disable the preference altogether if the kill switch is active. Next PR?